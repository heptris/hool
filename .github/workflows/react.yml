name: React CI

on:
  push:
    branches:
      - develop

jobs:
  build:
    runs-on: ubuntu-20.04

    steps:
      - uses: actions/checkout@v3

      - name: Create secrets
        run: echo "${{ secrets.FRONT_DOTENV_LOCAL }}" > frontend/.env.local

      - name: Set up Node.js 16.16
        uses: actions/setup-node@v3
        with:
          node-version: 16.16

      - name: Install dependencies
        run: |
          cd frontend/
          npm install

      # - name: Run tests
      #   run: |
      #     cd frontend/
      #     npm run test

      - name: Dockerize
        run: |
          echo "${{ secrets.FRONT_DOCKER_HUB_PW }}" | docker login -u "${{ secrets.FRONT_DOCKER_HUB_USER }}" --password-stdin
          docker build -t "${{ secrets.FRONT_IMAGE_NAME }}" ./frontend/
          docker push "${{ secrets.FRONT_IMAGE_NAME }}"
          docker logout
