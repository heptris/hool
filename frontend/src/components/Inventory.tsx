import { useState } from "react";

import styled from "styled-components";
import { darkTheme } from "styles/Theme";
import { letterSpacingSize } from "styles/GlobalStyle";

function Inventory() {
  const [isOwnItems, setIsOwnItems] = useState(true);
  const [isDisplayModal, setIsDisplayModal] = useState(false);

  const switchIsItems = () => {
    setIsOwnItems(!isOwnItems);
  };

  const switchIsDisplayModal = () => {
    setIsDisplayModal(!isDisplayModal);
  };

  return (
    <InventoryBox>
      <InventoryHeader>인벤토리</InventoryHeader>
      <InventorySwitches>
        <SwitchItem
          style={
            isOwnItems
              ? {
                  background: `linear-gradient(to bottom, transparent 85%, ${darkTheme.mainBadgeColor} 15%)`,
                }
              : {}
          }
          disabled={isOwnItems ? true : false}
          onClick={switchIsItems}
        >
          <span style={isOwnItems ? { color: darkTheme.mainBadgeColor } : {}}>
            소유중
          </span>
        </SwitchItem>
        <SwitchItem
          style={
            isOwnItems
              ? {}
              : {
                  background: `linear-gradient(to bottom, transparent 85%, ${darkTheme.mainBadgeColor} 15%)`,
                }
          }
          disabled={isOwnItems ? false : true}
          onClick={switchIsItems}
        >
          <span style={isOwnItems ? {} : { color: darkTheme.mainBadgeColor }}>
            찜
          </span>
        </SwitchItem>
      </InventorySwitches>
      <Hr />

      <InventoryContent></InventoryContent>
    </InventoryBox>
  );
}

const InventoryBox = styled.div`
  width: 60%;
  height: 90%;
  border: 1px solid ${darkTheme.adaptiveGrey700};
  border-radius: 4px;
  margin: 0 0 0 4rem;
  background-color: ${darkTheme.mainColor};
`;

const InventoryHeader = styled.div`
  font-size: 1.25rem;
  font-weight: bold;
  margin: 0.8rem 0 1.5rem 0.8rem;
`;

const InventorySwitches = styled.div`
  display: flex;
  margin: 0 0 0 0.8rem;
`;

const SwitchItem = styled.button`
  background-color: transparent;
  color: ${darkTheme.white};
  margin: 0 2rem 0 0;
  padding: 0 0 0.25rem 0;

  span {
    margin-right: ${letterSpacingSize}rem;
  }

  &:hover {
    cursor: pointer;

    span {
      color: ${darkTheme.adaptiveGrey200};
    }
  }
`;

const Hr = styled.hr`
  background-color: ${darkTheme.adaptiveGrey700};
  border: 1px solid ${darkTheme.adaptiveGrey700};
  margin: 0;
`;

const InventoryContent = styled.div`
  display: flex;
  width: 70%;
`;

export default Inventory;
