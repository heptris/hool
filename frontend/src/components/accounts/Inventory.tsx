import { useState } from "react";

import styled from "styled-components";
import { darkTheme } from "styles/Theme";
import { letterSpacingSize } from "styles/GlobalStyle";

function Inventory() {
  const [isOwnItems, setIsOwnItems] = useState(true);
  const [isDisplayModal, setIsDisplayModal] = useState(false);

  const switchIsDisplayModal = () => {
    setIsDisplayModal(!isDisplayModal);
  };

  const MyOwnItems = [];

  const MyFavorites = [];

  return (
    <InventoryBox>
      <InventoryHeader>인벤토리</InventoryHeader>
      <InventorySwitches>
        <SwitchItem
          style={
            isOwnItems
              ? {
                  background: `linear-gradient(to bottom, transparent 85%, ${darkTheme.mainBadgeColor} 15%)`,
                }
              : {}
          }
          onClick={() => {
            setIsOwnItems(true);
          }}
        >
          <span style={isOwnItems ? { color: darkTheme.mainBadgeColor } : {}}>
            소유중
          </span>
        </SwitchItem>
        <SwitchItem
          style={
            isOwnItems
              ? {}
              : {
                  background: `linear-gradient(to bottom, transparent 85%, ${darkTheme.mainBadgeColor} 15%)`,
                }
          }
          onClick={() => {
            setIsOwnItems(false);
          }}
        >
          <span style={isOwnItems ? {} : { color: darkTheme.mainBadgeColor }}>
            즐겨찾기
          </span>
        </SwitchItem>
      </InventorySwitches>
      <Hr />

      <InventoryContent></InventoryContent>
    </InventoryBox>
  );
}

const InventoryBox = styled.div`
  width: 100%;
  min-height: 45rem;
  border: 1px solid ${darkTheme.adaptiveGrey700};
  border-radius: 4px;
  margin: 0 0 0 4rem;
  background-color: ${darkTheme.mainColor};
`;

const InventoryHeader = styled.div`
  font-size: 1.25rem;
  font-weight: bold;
  margin: 0.8rem 0 1.5rem 0.8rem;
`;

const InventorySwitches = styled.div`
  display: flex;
  margin: 0 0 0 0.8rem;
`;

const SwitchItem = styled.button`
  background-color: transparent;
  color: ${darkTheme.white};
  margin: 0 2rem 0 0;
  padding: 0 0 0.25rem 0;

  span {
    margin-right: ${letterSpacingSize}rem;
  }

  &:hover {
    cursor: pointer;

    span {
      color: ${darkTheme.adaptiveGrey200};
    }
  }
`;

const Hr = styled.hr`
  background-color: ${darkTheme.adaptiveGrey700};
  border: 1px solid ${darkTheme.adaptiveGrey700};
  margin: 0;
`;

const InventoryContent = styled.div`
  display: flex;
  width: 70%;
`;

export default Inventory;
