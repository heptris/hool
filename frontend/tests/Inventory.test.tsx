import React from "react";
import { render } from "@testing-library/react";
import Inventory from "../src/components/Inventory";

describe("<Inventory />", () => {
  it("matches snapshot", () => {
    const utils = render(<Inventory />);
    expect(utils.container).toMatchSnapshot();
  });

  it("shows the props correctly", () => {
    const utils = render(<Inventory />);
    utils.getByText("인벤토리");
    utils.getByText("소유중");
    utils.getByText("찜");
  });
});
