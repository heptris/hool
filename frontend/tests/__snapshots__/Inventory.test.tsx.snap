// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inventory /> matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ kknEKR"
  >
    <div
      class="sc-gsnTZi gkOHpF"
    >
      인벤토리
    </div>
    <div
      class="sc-dkzDqf dQXOka"
    >
      <button
        class="sc-hKMtZM kbhiCc"
      >
        <span
          style="color: rgb(0, 198, 193);"
        >
          소유중
        </span>
      </button>
      <button
        class="sc-hKMtZM kbhiCc"
      >
        <span>
          즐겨찾기
        </span>
      </button>
    </div>
    <hr
      class="sc-eCYdqJ fmQJPI"
    />
    <div
      class="sc-jSMfEi bQlSLI"
    />
  </div>
</div>
`;
